### PR TITLE
Support Windows line endings

### PR DESCRIPTION
**Pull request by evhub**
Previously, since the file was opened as bytes but written as text, `\r\n` got written as `\r\r\n`, since it read in `\r\n` as bytes, passed through the `\r`, then wrote out `\r\n` as text, which using Python's universal newline support automatically transforms `\n` into `\r\n` on Windows, resulting in `\r\r\n`. Thus, this was adding bunches of `\r` characters every time it was run. This PR fixes that.

### CODE REVIEW

**Comment by asottile** [comment]
I imagine https://github.com/asottile/pyupgrade has the same issue.  Let's fix this in a different way as this will currently change newlines to whatever the platform picks instead of preserving whatever was in the file.

Since we're reading the correct bytes, it really just means that we need to write out the correct bytes.

We use the same trick in pre-commit/pre-commit-hooks:

```python
with io.open(filename, 'w', encoding='UTF-8', newline='') as f:
    f.write(...)
```

The `newline=''` kwarg tells python to not touch up line endings at all.

Also your fix introduces a bug:
```pycon
>>> with open('f.txt', 'wb') as f:
...     for _ in range(1000):
...         f.write(u'Hello euro: €€€€€\n'.encode('UTF-8'))
...     f.write(u'€'.encode('cp1252'))
... 
>>> import io
>>> io.open('f.txt')
<_io.TextIOWrapper name='f.txt' encoding='UTF-8'>
>>> io.open('f.txt').read()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/codecs.py", line 314, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf8' codec can't decode byte 0x80 in position 28000: invalid start byte
```

Can you also add an appveyor configuration?

This is the minimal configuration: https://github.com/pre-commit/pre-commit-hooks/blob/master/appveyor.yml

I'll enable it once you've added it to your branch :)

**Comment by evhub** [comment]
@asottile Does this look good to you?

File: add_trailing_comma.py [hunk 1]
```diff
@@ -363,7 +363,7 @@ def _fix_src(contents_text, py35_plus, py36_plus):
 
 
 def fix_file(filename, args):
-    with open(filename, 'rb') as f:
+    with io.open(filename, 'rb') as f:
```
Comment (asottile) [inline add_trailing_comma.py]: this line shouldn't change

Comment (evhub) [inline add_trailing_comma.py]: Done.

**Comment by evhub** [comment]
@asottile Regression test added and passing.

**Comment by asottile** [comment]
This is available in 0.6.3

**Comment by asottile** [comment]
I've made a similar patch to pyupgrade [here](https://github.com/asottile/pyupgrade/pull/20)

**Pull request merged**